### ISSUE TITLE
Deprecated class Text of module typing

### ISSUE THREAD

**Issue by RequieMa** [microsoft/qlib#1815]
## 🐛 Bug Description

![image](https://github.com/microsoft/qlib/assets/57754488/64bd985c-a78b-409d-8695-cd7bd86ee73d)

Is this type of pylint suggestion required to consider from your perspective?

**Comment by SunsetWolf** [comment]
Hi, I pulled the latest code, ran the pylint command, and did not find the problem you describe. I found the error code `R0201` in your command, which is not present in the latest code.

The command I used was:
* The hit for testing the complete qlib folder: https://github.com/microsoft/qlib/blob/main/.github/workflows/test_qlib_from_source.yml#L109
* For the gbdt.py file: `pylint --disable=C0104,C0114,C0115,C0116,C0301,C0302,C0411,C0413,C1802,R0401,R0801,R0902,R0903,R0911,R0912,R0913,R0914,R0915,R1720,W0105,W0123,W0201,W0511,W0613,W1113,W1514,E0401,E1121,C0103,C0209,R0402,R1705,R1710,R1725,R1735,W0102,W0212,W0221,W0223,W0231,W0237,W0612,W0621,W0622,W0703,W1309,E1102,E1136 --const-rgx='[a-z_][a-z0-9_]{2,30}$' qlib/contrib/model/gbdt.py --init-hook "import astroid; astroid.context.Inferenc
eContext.max_inferred = 500; import sys; sys.setrecursionlimit(2000)"`

**Comment by RequieMa** [comment]
Quite weird. I'm using the latest version
![image](https://github.com/microsoft/qlib/assets/57754488/67216c50-aec0-4c7e-90eb-3d279caf8b7f)
I tried with your command. It gave a different error code `W4904`, but the same reason 'deprecated-class'

**Comment by qew21** [comment]
`typing.Text` can be replaced by `str` and was deprecated since python 3.11 according to [gh-92332](https://github.com/python/cpython/issues/92332).

**Comment by SunsetWolf** [comment]
> Quite weird. I'm using the latest version ![image](https://private-user-images.githubusercontent.com/57754488/342734102-67216c50-aec0-4c7e-90eb-3d279caf8b7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0NjUwODcsIm5iZiI6MTcxOTQ2NDc4NywicGF0aCI6Ii81Nzc1NDQ4OC8zNDI3MzQxMDItNjcyMTZjNTAtYWVjMC00YzdlLTkwZWItM2QyNzljYWY4YjdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI3VDA1MDYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNlNWU5YWUyYTQyNjhjOGFjZDM2YjgwN2IzYTc5ZmRiYzk4ZjEyYmFjMzgwNzE1NDJhNWI3ZmJhYzU0ZTM1ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4F_lroRFZvMU6gJBO8yX3QYyHHol7gH8b1BJ4JdT52s) I tried with your command. It gave a different error code `W4904`, but the same reason 'deprecated-class'

I tested `gbdt.py` after upgrading `pylint` to `3.2.4` and found no problems.

**Comment by SunsetWolf** [comment]
I don't think it makes sense to test with python 3.11, since qlib currently supports the following python versions: [3.7 and 3.8](https://github.com/microsoft/qlib/blob/main/.github/workflows/test_qlib_from_source.yml#L22).
